lib/schema-controller.js

```diff
@@ -0,0 +1,169 @@
+'use strict'
+
+const { Schemas } = require('./schemas')
+
+const Ajv = require('ajv')
+const fastJsonStringify = require('fast-json-stringify')
+
+let buildPerformanceValidator
+
+function SchemaController () {
+  this.schemaBucket = new Schemas()
+  this.validatorCompiler = undefined
+  this.serializerCompiler = undefined
+  this.parent = undefined
+}
+
+/**
+ * This method is called once on the fastify server inizialization.
+ * You may use it to set some globals argument as caching
+ */
+SchemaController.prototype.setup = function () {
+  // it is needed once at the server built
+  buildPerformanceValidator = ValidatorSelector()
+}
+
+/**
+ * Called at every fastify context that is being created.
+ * This should return a new SchemaController.
+ * WARN: If it returns `this`, the JSON schema encapsulation will be broken
+ */
+SchemaController.prototype.clone = function clone () {
+  const wrapper = new SchemaController()
+
+  const clonedSchema = new Schemas()
+  Object.values(this.schemaBucket.getSchemas())
+    .forEach(_ => clonedSchema.add(_))
```

Comment:
Can we defer creating this until some new schema is added? I would like to keep the memory footprint low.

Reply:
I will try it 👍